### PR TITLE
LibWeb: Implement navigation.reload, traverseTo, back, and forwards

### PR DESCRIPTION
**Pull request by ADKaster**
The proper abstract operations on Navigable and TraversableNavigable are
not quite ready to call from Navigation. With this patch all of the
user-facing APIs of Navigation are in place, and the stage should be set
to implement the parts of the navigation and traversal AOs that need to
interact with the Navigation object.

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Navigation.cpp [hunk 1]
```diff
@@ -322,6 +323,62 @@ WebIDL::ExceptionOr<NavigationResult> Navigation::reload(NavigationReloadOptions
...
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-forward
+WebIDL::ExceptionOr<NavigationResult> Navigation::forward(NavigationOptions const& options)
+{
+    auto& realm = this->realm();
+    // The forward(options) method steps are:
+
+    // 1. If this's current entry index is −1 or 0, then return an early error result for an "InvalidStateError" DOMException.
+    if (m_current_entry_index == -1 || m_current_entry_index == static_cast<i64>(m_entry_list.size()))
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/Navigation.cpp:372]: The comment here doesn't match the code :)

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/Navigation.cpp]: ah, forgot to properly copy over the spec text when copy pasta the implementation to tweak it.
https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-forward
Should be all set now.

File: Userland/Libraries/LibWeb/HTML/Navigation.cpp [hunk 2]
```diff
@@ -322,6 +323,63 @@ WebIDL::ExceptionOr<NavigationResult> Navigation::reload(NavigationReloadOptions
...
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-forward
+WebIDL::ExceptionOr<NavigationResult> Navigation::forward(NavigationOptions const& options)
+{
+    auto& realm = this->realm();
+    // The forward(options) method steps are:
+
+    // 1. If this's current entry index is −1 or is equal to this's entry list's size − 1,
+    //    then return an early error result for an "InvalidStateError" DOMException.
+    if (m_current_entry_index == -1 || m_current_entry_index == static_cast<i64>(m_entry_list.size()))
```
Comment (kalenikaliaksandr) [inline Userland/Libraries/LibWeb/HTML/Navigation.cpp:373]: "equal to this's entry list's size − 1" looks like you missed -1 part

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/Navigation.cpp]: do'h I sure did! fixed now

**Review by awesomekling — APPROVED** [review]

**Pull request merged**